core/src/display_object/avm1_button.rs

```diff
@@ -399,8 +401,33 @@ impl<'gc> TDisplayObject<'gc> for Avm1Button<'gc> {
         true
     }
 
-    fn on_focus_changed(&self, gc_context: &Mutation<'gc>, focused: bool) {
-        self.0.write(gc_context).has_focus = focused;
+    fn on_focus_changed(
+        &self,
+        context: &mut UpdateContext<'_, 'gc>,
+        focused: bool,
+        other: Option<DisplayObject<'gc>>,
+    ) {
+        self.0.write(context.gc_context).has_focus = focused;
+
+        if let Value::Object(object) = self.object() {
```

Comment:
Could you extract the body of this `if let` into a separate function, and call it from all of the displayobjects that need it?